xdiff/xdiff.php

```diff
@@ -0,0 +1,200 @@
+ * @param string $new_file Path to the second file. This file acts as "new" file.
+ * @param string $dest Path of the resulting patch file. Resulting file contains differences between
+ * "old" and "new" files. It is in binary format and is human-unreadable.
+ * @return bool true on success or false on failure.
+ */
+function xdiff_file_diff_binary(string $old_file, string $new_file, string $dest): bool {}
+
+/**
+ * Make unified diff of two files
+ * @link https://www.php.net/manual/en/function.xdiff-file-diff.php
+ * @param string $old_file Path to the first file. This file acts as "old" file.
+ * @param string $new_file Path to the second file. This file acts as "new" file.
+ * @param string $dest Path of the resulting patch file.
+ * @param int $context Indicates how many lines of context you want to include in diff result.
+ * @param bool $minimal Set this parameter to true if you want to minimalize size of the result (can take a long time).
+ * @return bool true on success or false on failure.
+ */
+function xdiff_file_diff(
+    string $old_file,
+    string $new_file,
+    string $dest,
+    int $context = 3,
+    bool $minimal = false
+): bool {}
+
+/**
+ * Merge 3 files into one
+ * @link https://www.php.net/manual/en/function.xdiff-file-merge3.php
+ * @param string $old_file Path to the first file. It acts as "old" file.
+ * @param string $new_file1 Path to the second file. It acts as modified version of old_file.
+ * @param string $new_file2 Path to the third file. It acts as modified version of old_file.
+ * @param string $dest Path of the resulting file, containing merged changed from both new_file1 and new_file2.
+ * @return mixed true if merge was successful, string with rejected chunks if it was not or false if an internal
+ * error happened.
+ */
+function xdiff_file_merge3(string $old_file, string $new_file1, string $new_file2, string $dest) {}
+
+/**
+ * Alias of xdiff_file_bpatch
+ * @link https://www.php.net/manual/en/function.xdiff-file-patch-binary.php
+ * @param string $file The original file.
+ * @param string $patch The binary patch file.
+ * @param string $dest Path of the resulting file.
+ * @return bool true on success or false on failure.
+ */
+function xdiff_file_patch_binary(string $file, string $patch, string $dest): bool {}
+
+/**
+ * Patch a file with an unified diff
+ * @link https://www.php.net/manual/en/function.xdiff-file-patch.php
+ * @param string $file The original file.
+ * @param string $patch The unified patch file. It has to be created using xdiff_string_diff(), xdiff_file_diff()
+ * functions or compatible tools.
+ * @param string $dest Path of the resulting file.
+ * @param int $flags Can be either XDIFF_PATCH_NORMAL (default mode, normal patch) or XDIFF_PATCH_REVERSE
+ * (reversed patch). Starting from version 1.5.0, you can also use binary OR to enable XDIFF_PATCH_IGNORESPACE flag.
+ * @return mixed false if an internal error happened, string with rejected chunks if patch couldn't be applied
```

Comment:
It makes sense to change `mixed` to `bool|string` to make type inference and inspections more correct and strict